pennylane/tape/tape.py

```diff
@@ -237,6 +237,47 @@ def stop_at(obj):  # pylint: disable=unused-argument
     return new_tape
 
 
+def expand_tape_state_prep(tape, skip_first=True):
+    """Expand all instances of StatePrep operations in the tape.
+
+    Args:
+        tape (Union[QuantumTape, QuantumScript]): The tape to expand.
+        skip_first (Bool): If ``True``, will not expand a StatePrep operation if
+            it is the first operation in the tape.
+
+    Returns:
+        QuantumTape: The expanded version of ``tape``.
+
+    **Example**
+    ...
+    """
+    new_prep = []
+    new_ops = []
+
+    for i, op in enumerate(tape._prep + tape._ops):
+        if isinstance(op, StatePrep):
+            if i == 0 and skip_first:
```

Comment:
Since we only care about `i==0 and skip_first`, maybe we could handle this case before entering the loop?

Reply:
Nice catch! added 👍🏼